File: ruby/lib/k256.rb
# frozen_string_literal: true

require_relative "k256/version"
require_relative "k256/types"
require_relative "k256/base58"
require_relative "k256/decoder"
require_relative "k256/client"

# K256 SDK - The gateway to Solana's liquidity ecosystem
#
# @example
#   require 'k256'
#
#   client = K256::Client.new(api_key: ENV['K256_API_KEY'])
#
#   client.on_pool_update do |update|
#     puts "Pool #{update.pool_address}: slot=#{update.slot}"
#   end
#
#   client.connect
#   client.subscribe(channels: ['pools', 'priority_fees', 'blockhash'])
module K256
  class Error < StandardError; end
  class ConnectionError < Error; end
  class DecodeError < Error; end
end

File: go/k256.go
// Package k256 provides the official Go SDK for K256 - the gateway to Solana's liquidity ecosystem.
//
// Connect any application to Solana's liquidity ecosystem. One API. All venues. Full observability.
//
// Quick Start:
//
//	package main
//
//	import (
//		"fmt"
//		k256 "github.com/k256-xyz/k256-sdks/go"
//	)
//
//	func main() {
//		ws := k256.NewWebSocket(k256.Config{
//			APIKey: "your-api-key",
//		})
//		defer ws.Close()
//
//		ws.OnPoolUpdate(func(update *k256.PoolUpdate) {
//			fmt.Printf("Pool %s: slot=%d\n", update.PoolAddress, update.Slot)
//		})
//
//		if err := ws.Connect(); err != nil {
//			panic(err)
//		}
//
//		ws.Subscribe(k256.SubscribeRequest{
//			Channels: []string{"pools", "priority_fees", "blockhash"},
//		})
//
//		select {}
//	}
package k256

// Version is the current SDK version.
const Version = "0.1.0"
